src/core/src/pass/manager.cpp

```diff
@@ -89,89 +130,184 @@ class stopwatch {
-        needs_validate = pass_applied;
     }
-    if (profile_enabled) {
-        cout << "passes done in " << overall_timer.get_milliseconds() << "ms\n";
+
+private:
+    static std::string gen_file_name(const std::string& model_name, const std::string& pass_name, const size_t idx) {
+        std::stringstream name;
+        // visualizations and serializations will be named after the outermost function
+        const size_t num_digits_in_pass_index = 3;
+        std::string index_str = std::to_string(idx);
+        index_str = std::string(num_digits_in_pass_index - index_str.length(), '0') + index_str;
+
+        name << model_name << std::string("_") << index_str << std::string("_") << pass_name;
+        return name.str();
     }
 
-    return function_changed;
+    std::unordered_map<std::string, stopwatch> stopwatches;
+
+    EnvVar m_visualize;
+    EnvVar m_serialize;
+    EnvVar m_profile_pass;
+
+    std::string m_manager_name;
+    std::fstream m_file;
+};
+
+}  // namespace
+
+ov::pass::Manager::Manager() : m_pass_config(std::make_shared<PassConfig>()) {}
+
+ov::pass::Manager::~Manager() = default;
+
+ov::pass::Manager::Manager(std::string name) : m_pass_config(std::make_shared<PassConfig>()), m_name(std::move(name)) {}
+
+ov::pass::Manager::Manager(std::shared_ptr<ov::pass::PassConfig> pass_config, std::string name)
+    : m_pass_config(std::move(pass_config)),
+      m_name(std::move(name)) {}
+
+void ov::pass::Manager::set_per_pass_validation(bool new_state) {
+    m_per_pass_validation = new_state;
+}
+
+bool ov::pass::Manager::run_passes(const shared_ptr<ov::Model>& model) {
+    OV_ITT_SCOPED_TASK(ov::itt::domains::core, "pass::Manager::run_passes");
+    Profiler profiler(m_name);
+
+    bool model_changed = false;
+    bool pass_changed_model = false;
+
+    profiler.start_timer(m_name);
+    for (const auto& pass : m_pass_list) {
+        const auto& pass_name = pass->get_name();
+
+        profiler.start_timer(pass_name);
+        pass_changed_model = run_pass(pass, model, pass_changed_model);
+        profiler.stop_timer(pass_name, pass_changed_model);
+
+        model_changed = model_changed || pass_changed_model;
+
+        profiler.visualize(model, pass_name);
+        profiler.serialize(model, pass_name);
+    }
+    profiler.stop_timer(m_name, model_changed);
+
+    return model_changed;
+}
+
+bool ov::pass::Manager::run_pass(const std::shared_ptr<PassBase>& pass,
+                                 const std::shared_ptr<Model>& model,
+                                 bool needs_validate) {
+    if (m_pass_config->is_disabled(pass->get_type_info())) {
+        OPENVINO_DEBUG("Pass ", pass->get_name(), " is disabled.");
+        return false;
+    }
+
+    // This checks if we need to skip the graph transformation when the graph pass relies on
+    // static shape but the model state is dynamic.
+    if (pass->get_property(PassProperty::REQUIRE_STATIC_SHAPE) && model->is_dynamic()) {
```

Comment:
Do we use it in the project? Asking out of curiosity.

Reply:
to be honest, I don't know
most probably we don't , I will try to figure it out and might be delete it in the next PR